java-checks/src/main/java/org/sonar/java/checks/unused/UnusedPrivateMethodCheck.java

```diff
@@ -90,22 +96,15 @@ private void reportUnusedPrivateMethods() {
 
   @Override
   public void visitNode(Tree tree) {
-    switch (tree.kind()) {
-      case METHOD:
-      case CONSTRUCTOR:
-        checkIfUnused((MethodTree) tree);
-        break;
-      case NEW_CLASS:
-        checkIfUnknown((NewClassTree) tree);
-        break;
-      case METHOD_INVOCATION:
-        checkIfUnknown((MethodInvocationTree) tree);
-        break;
-      case METHOD_REFERENCE:
-        checkIfUnknown((MethodReferenceTree) tree);
-        break;
-      default:
-        throw new IllegalStateException("Unexpected subscribed tree.");
+    Kind kind = tree.kind();
```

Comment:
to be honest I feel it was much nicer and easier to read before... I would not challenge this and be OK with the line not being covered. Can you revert the change?
You can also decide to change the last default case to not throw and just do nothing instead... will still remove one line of code.

```java
     default:
        // do nothing
    }